lib/Ebook.php

```diff
@@ -797,23 +1383,318 @@ private function NullIfEmpty($elements): ?string{
+			GitHubUrl = ?,
+			WikipediaUrl = ?,
+			EbookCreated = ?,
+			EbookUpdated = ?,
+			TextSinglePageByteCount = ?,
+			IndexableText = ?
+			where
+			EbookId = ?
+		', [$this->Identifier, $this->WwwFilesystemPath, $this->RepoFilesystemPath, $this->KindleCoverUrl, $this->EpubUrl,
+				$this->AdvancedEpubUrl, $this->KepubUrl, $this->Azw3Url, $this->DistCoverUrl, $this->Title,
+				$this->FullTitle, $this->AlternateTitle, $this->Description, $this->LongDescription,
+				$this->Language, $this->WordCount, $this->ReadingEase, $this->GitHubUrl, $this->WikipediaUrl,
+				$this->EbookCreated, $this->EbookUpdated, $this->TextSinglePageByteCount, $this->IndexableText,
+				$this->EbookId]);
+
+		$this->DeleteTags();
+		$this->InsertTags();
+
+		$this->DeleteLocSubjects();
+		$this->InsertLocSubjects();
+
+		$this->DeleteCollectionMemberships();
+		$this->InsertCollectionMemberships();
+
+		$this->DeleteGitCommits();
+		$this->InsertGitCommits();
+
+		$this->DeleteSources();
+		$this->InsertSources();
+
+		$this->DeleteContributors();
+		$this->InsertContributors();
+
+		$this->DeleteTocEntries();
+		$this->InsertTocEntries();
+	}
+
+	private function DeleteTags(): void{
+		Db::Query('
+			DELETE from EbookTags
+			where
+			EbookId = ?
+		', [$this->EbookId]
+		);
+	}
+
+	private function InsertTags(): void{
+		foreach($this->Tags as $tag){
+			Db::Query('
+				INSERT into EbookTags (EbookId, TagId)
+				values (?,
+				        ?)
+			', [$this->EbookId, $tag->TagId]);
 		}
+	}
+
+	private function DeleteLocSubjects(): void{
+		Db::Query('
+			DELETE from EbookLocSubjects
+			where
+			EbookId = ?
+		', [$this->EbookId]
+		);
+	}
+
+	private function InsertLocSubjects(): void{
+		foreach($this->LocSubjects as $locSubject){
+			Db::Query('
+				INSERT into EbookLocSubjects (EbookId, LocSubjectId)
+				values (?,
+				        ?)
+			', [$this->EbookId, $locSubject->LocSubjectId]);
+		}
+	}
+
+	private function DeleteCollectionMemberships(): void{
+		Db::Query('
+			DELETE from CollectionEbooks
+			where
+			EbookId = ?
+		', [$this->EbookId]
+		);
+	}
+
+	private function InsertCollectionMemberships(): void{
+		foreach($this->CollectionMemberships as $collectionMembership){
+			$collectionMembership->EbookId = $this->EbookId;
+			$collectionMembership->CollectionId = $collectionMembership->Collection->CollectionId;
+
+			Db::Query('
+				INSERT into CollectionEbooks (EbookId, CollectionId, SequenceNumber)
+				values (?,
+					?,
+				        ?)
+			', [$collectionMembership->EbookId, $collectionMembership->CollectionId, $collectionMembership->SequenceNumber]);
+
+			$collectionMembership->CollectionEbookId = Db::GetLastInsertedId();
+		}
+	}
 
-		return false;
+	private function DeleteGitCommits(): void{
+		Db::Query('
+			DELETE from GitCommits
+			where
+			EbookId = ?
+		', [$this->EbookId]
+		);
+	}
+
+	private function InsertGitCommits(): void{
+		foreach($this->GitCommits as $commit){
+			$commit->EbookId = $this->EbookId;
+
+			Db::Query('
```

Comment:
This should be in `GitCommit::Create()` not in `Ebook`

Reply:
Done.